crates/router/src/core/payments.rs

```diff
@@ -2023,11 +2023,50 @@ pub enum CallConnectorAction {
     HandleResponse(Vec<u8>),
 }
 
+#[derive(Clone, Default, Debug)]
+struct Private;
+
 #[derive(Clone, Default, Debug)]
 pub struct PaymentAddress {
-    pub shipping: Option<api::Address>,
-    pub billing: Option<api::Address>,
-    pub payment_method_billing: Option<api::Address>,
+    shipping: Option<api::Address>,
+    billing: Option<api::Address>,
+    payment_method_billing: Option<api::Address>,
+    // To restrict direct construction of PaymentAddress
+    _private: Private,
```

Comment:
This would still allow us to construct PaymentAddress within this file right? Is there any way to avoid that too?

Reply:
Yes, will have to move this to a module, will make the necessary changes